biorxiv/article_distances/biorxiv_corrected_published_rate.py

```diff
@@ -0,0 +1,520 @@
+            {"month":"2018-09", "document":"423517_v1.xml"},
+            {"month":"2019-01", "document":"528299_v1.xml"}
+        ]
+    )
+else:
+    published_preprint_df = pd.read_csv(
+        file_path, 
+        sep="\t"
+    )
+print(published_preprint_df.shape)
+published_preprint_df.head()
+
+
+# In[14]:
+
+
+if not file_path.exists():
+    final_mapped_df = (
+        published_preprint_df
+        .merge(
+            updated_biorxiv_journal_df, 
+            on="document"
+        )
+        .rename(index=str, columns={"month":"pub_month"})
+    )
+    print(final_mapped_df.shape)
+else:
+    final_mapped_df = published_preprint_df
+    print(final_mapped_df.shape)
+
+final_mapped_df
+
+
+# In[15]:
+
+
+if not file_path.exists():
+    (
+        final_mapped_df
+        .to_csv(
+            "output/biorxiv_articles_published_date.tsv",
+            sep="\t", index=False
+        )
+    )
+
+
+# In[16]:
+
+
+snapshot_wo_links_df = (
+    biorxiv_journal_df
+    .assign(published_closed=lambda x: (x.pmcoa != x.published))
+    .merge(
+        final_mapped_df[["document", "pub_month"]],
+        on="document"
+    )
+    .replace({"2013-10":"2013-11"})
+    .groupby("pub_month")
+    .agg({
+        "published":lambda x: (x==True).sum(),
+        "pub_month":"count",
+        "published_closed":lambda x: (x==True).sum(),
+    })
+    .rename(index=str, columns={"pub_month":"posted"})
+    .reset_index()
+    .assign(
+        label="2020 Snapshot",
+        rate=lambda x: x.published/x.posted
+    )
+)
+snapshot_wo_links_df.head()
+
+
+# In[17]:
+
+
+snapshot_w_links_df = (
+    final_mapped_df
+    .replace({"2013-10":"2013-11"})
+    .groupby("pub_month")
+    .agg({
+        "published":lambda x: (x==True).sum(),
+        "pub_month":"count"
+    })
+    .rename(index=str, columns={"pub_month":"posted"})
+    .reset_index()
+    .assign(
+        rate=lambda x: x.published/x.posted,
+        label="2020 Snapshot+Missing Links"
+    )
+)
+snapshot_w_links_df.head()
+
+
+# ## Calculate the publication rate
+
+# In[18]:
+
+
+publish_rate_df = (
+    snapshot_w_links_df
+    .append(
+        abdill_df
+         .assign(label="Abdill et al. (2018)")
+         .rename(index=str, columns={"month":"pub_month"}),
+        sort=False
+    )
+    .append(snapshot_wo_links_df.drop("published_closed", axis=1))
+    .reset_index(drop=True)
+)
+publish_rate_df.sample(10, random_state=100)
+
+
+# In[19]:
+
+
+publish_rate_df['pub_month'] = (
+    pd.Categorical(
+        publish_rate_df.pub_month.values.tolist(), 
+        ordered=True
+    )
+)
+
+posted = (
+    publish_rate_df
+    .query("label=='2020 Snapshot+Missing Links'")
+    .posted
+    .sum()
+)
+
+published = (
+    publish_rate_df
+    .query("label=='2020 Snapshot+Missing Links'")
+    .published
+    .sum()
+)
+print(f"Published: {published}")
+print(f"Posted: {posted}")
+print(f"Overall proportion published: {published/posted:.4f}")
+
+
+# In[20]:
+
+
+if False:
```

Comment:
Is this a faster way of commenting out code? If so, do you still need these three cells?

Reply:
> Is this a faster way of commenting out code? If so, do you still need these three cells? 

This code here was being used to try and estimate the true number of missing biorxiv-pmc links; however, it was determined that the equation isn't needed anymore. I meant to delete it but forgot.